### DISCUSSION TITLE
How to render multiple tables with subcaptions using kable() and a for loop?

### DISCUSSION THREAD

**Discussion by kmink3225** [quarto-dev/quarto-cli#5935] [Q&A] [answered]
### Description

# Description

Hello,

I am doing some project at company. But It seems that it is not easy to properly use Python with many cross-references, especially for sub captions both in making multiple tables and plots. So, I decided to use R. 

My question is how to render multiple tables with subcaptions using kable() and a for loop? It seems that there is no function like display() of Python in R. For example,

````qmd

---
format: html
engine: knitr
execute:
  warning: false
---

```{r}
rm(list=ls())
library(tidyverse)
library(knitr)
```


```{r working}
#| label: tbl-mtcars2
#| tbl-cap: "Mtcars Data"
#| tbl-subcap: 
#|     - "A"
#|     - "B"
#|     - "C"
#|     - "D"
#|     - "E"
#|     - "F"

mtcars %>% filter(mtcars$carb==unique(mtcars$carb)[1]) %>%kable()
mtcars %>% filter(mtcars$carb==unique(mtcars$carb)[2]) %>%kable()
mtcars %>% filter(mtcars$carb==unique(mtcars$carb)[3]) %>%kable()
mtcars %>% filter(mtcars$carb==unique(mtcars$carb)[4]) %>%kable()
mtcars %>% filter(mtcars$carb==unique(mtcars$carb)[5]) %>%kable()
mtcars %>% filter(mtcars$carb==unique(mtcars$carb)[6]) %>%kable()

```

```{r not_working}
#| label: tbl-mtcars
#| tbl-cap: "Mtcars Data"
#| tbl-subcap: 
#|     - "A"
#|     - "B"
#|     - "C"
#|     - "D"
#|     - "E"
#|     - "F"

temp_function<-function(i){
  temp=mtcars %>% filter(mtcars$carb==unique(mtcars$carb)[i])
  return(kable(temp))
}


for (i in 1:6){
 temp_function(i) 
}

# or

for (i in 1:6){
 temp=mtcars %>% filter(mtcars$carb==unique(mtcars$carb)[i])
 kable(temp)
}


```


````

# Output

![image](https://github.com/quarto-dev/quarto-cli/assets/61118160/dafc07f0-e3c8-4dbe-a2f6-94923badfaa1)

**Comment by mcanouil** [comment, marked as answer]
The following blog post should answer this: <https://mickael.canouil.fr/posts/2023-03-05-quarto-auto-table-crossref/>.

**Comment by kmink3225** [comment]
Hello @mcanouil,

I forgot to reply to your answer.
I visited your blog and tried your method. I was able to generate many tables using your code!
I will refer to your blog a lot. 
Thanks as always for your help!
